### ISSUE TITLE
Error while opening encoder for output stream -- Transcoding VAAPI

### ISSUE THREAD

**Issue by An4cl3t0** [jellyfin/jellyfin#1428]
Can't reproduce files after enabling transcoding VAAPI on Intel Graphics. Only tested MKV. The system does not start to reproduce de file and it "buffers" forever.

**Logs**
I reproduced the FFmpeg command with:
`/usr/bin/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -vaapi_device /dev/dri/renderD128 -i file:"VIDEOFILE.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 h264_vaapi  -b:v 7552001 -maxrate 7552001 -bufsize 15104002 -profile:v high -level 41 -force_key_frames "expr:gte(t,n_forced*3)" -vf "format=nv12|vaapi,hwupload" -copyts -vsync -1 -codec:a:0 aac -strict experimental -ac 2 -ab 384000 -af "volume=2" -f segment -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -segment_time 3  -individual_header_trailer 0 -segment_format mpegts -segment_list_type m3u8 -segment_start_number 0 -segment_list "/config/transcoding-temp/f409466ace839b054f56d2cf31a3143e.m3u8" -y "/config/transcoding-temp/f409466ace839b054f56d2cf31a3143e%d.ts"`

And it's output is:
```
[h264_vaapi @ 0x55bbc7143b60] [Eval @ 0x7ffcfe6d5150] Undefined constant or missing '(' in 'high'
[h264_vaapi @ 0x55bbc7143b60] Unable to parse option value "high"
[h264_vaapi @ 0x55bbc7143b60] Error setting option profile to value high.
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (ac3 (native) -> aac (native))
Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
```

**System**
 - OS: Docker
 - Browser: all
 - Jellyfin Version: 10.3.3

Thanks.

**Comment by sparky8251** [comment]
We are going to need someone to dig into this issue and find out what is going on and how we can solve it. I've tagged this with help wanted to let folks know. Hopefully someone takes up the mantle!

**Comment by GoodOmens83** [comment]
I've run into a similiar issue with my setup.

I'm attempting to use HW (VA API) Acceleration in docker and can't get it to work. I'm running the latest docker build (10.3.7) and running Docker 18.09.1 on a older Chrome Box (Running Libreelec) with a  Intel Celeron 2955U

Any insight on what I'm doing wrong? Settings are "VA API" with the default /dev/dri/renderD128

I've tried the following command to initiate my container:
`docker run -d --name jellyfin -v /storage/jellyfin/:/config -v /storage/jellyfin/cache:/cache -v /storage/video:/video -v /storage/transcode:/transcode --device /dev/dri/renderD128:/dev/dri/renderD128 --net=host -e ServerIP=$(ip route get 9.9.9.9 | awk '{ print $NF; exit }') --restart=unless-stopped --privileged jellyfin/jellyfin`

Installing and running vainfo within the docker shows the following:
```
error: XDG_RUNTIME_DIR not set in the environment.
error: can't connect to X server!
libva info: VA-API version 1.1.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_1_1
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.1 (libva 2.1.0)
vainfo: Driver version: Intel i965 driver for Intel(R) Haswell Mobile - 2.1.0
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264MultiviewHigh      : VAEntrypointVLD
      VAProfileH264StereoHigh         : VAEntrypointVLD
      VAProfileVC1Simple              : VAEntrypointVLD
      VAProfileVC1Main                : VAEntrypointVLD
      VAProfileVC1Advanced            : VAEntrypointVLD
      VAProfileNone                   : VAEntrypointVideoProc
      VAProfileJPEGBaseline           : VAEntrypointVLD
```

And finally here is the the output of the transcode:
```

https://xxxx.com:8920/emby/videos/09aba43d-0f90-e438-8684-d3016fe369db/hls1/main/0.ts?DeviceId=TW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NDsgcnY6NjguMCkgR2Vja28vMjAxMDAxMDEgRmlyZWZveC82OC4wfDE1NjM5MDMwOTgzNzM1&MediaSourceId=09aba43d0f90e4388684d3016fe369db&VideoCodec=h264&AudioCodec=aac&AudioStreamIndex=1&VideoBitrate=808001&AudioBitrate=192000&PlaySessionId=7d23d8c564b74b27b209eb6b39389559&api_key=815d4aa143ea4674a4d0188282089834&TranscodingMaxAudioChannels=2&RequireAvc=false&Tag=d5b69beaf03e413787b7c31a86fcf523&SegmentContainer=ts&MinSegments=1&BreakOnNonKeyFrames=True&h264-profile=high,main,baseline,constrainedbaseline&h264-level=51&TranscodeReasons=ContainerBitrateExceedsLimit

{"Protocol":"File","Id":"09aba43d0f90e4388684d3016fe369db","Path":"/video/Movies/movie.mkv","Type":"Default","Container":"mkv,webm","Size":995585920,"Name":"Blonde Ransom","IsRemote":false,"ETag":"d5b69beaf03e413787b7c31a86fcf523","RunTimeTicks":42005958656,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"RequiresClosing":false,"RequiresLooping":false,"SupportsProbing":true,"VideoType":"VideoFile","MediaStreams":[{"Codec":"mpeg2video","Language":"eng","TimeBase":"1/1000","CodecTimeBase":"1001/30000","VideoRange":"SDR","DisplayTitle":"480P MPEG2VIDEO","IsInterlaced":false,"BitRate":1896085,"RefFrames":1,"IsDefault":false,"IsForced":false,"Height":480,"Width":720,"AverageFrameRate":29.97003,"RealFrameRate":29.97003,"Profile":"Main","Type":"Video","AspectRatio":"4:3","Index":0,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"PixelFormat":"yuv420p","Level":8},{"Codec":"ac3","Language":"eng","TimeBase":"1/1000","CodecTimeBase":"1/48000","Title":"Stereo","DisplayTitle":"Eng Dolby Digital stereo Default","IsInterlaced":false,"ChannelLayout":"stereo","BitRate":192000,"Channels":2,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Type":"Audio","Index":1,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Level":0}],"Formats":[],"Bitrate":2088085,"RequiredHttpHeaders":{}}

/usr/local/bin/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -vaapi_device /dev/dri/renderD128 -i file:"/video/Movies/movie.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 h264_vaapi  -b:v 808001 -maxrate 808001 -bufsize 1616002 -profile:v high -level 41 -force_key_frames "expr:gte(t,n_forced*3)" -vf "format=nv12|vaapi,hwupload,scale_vaapi=w=640:h=426" -copyts -vsync -1 -codec:a:0 aac -strict experimental -ac 2 -ab 192000  -f segment -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -segment_time 3  -individual_header_trailer 0 -segment_format mpegts -segment_list_type m3u8 -segment_start_number 0 -segment_list "/transcode/transcoding-temp/19a6ef906eee56edfe9b441e5d10be20.m3u8" -y "/transcode/transcoding-temp/19a6ef906eee56edfe9b441e5d10be20%d.ts"


ffmpeg version 4.0.3 Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.3.0-27ubuntu1~18.04)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-vaapi --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gnutls --enable-gpl --enable-libass --enable-libfreetype --enable-libvidstab --enable-libmp3lame --enable-libopenjpeg --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx265 --enable-libxvid --enable-libx264 --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-postproc --enable-cuvid --enable-nvenc --enable-version3 --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib --extra-libs=-ldl --prefix=/opt/ffmpeg
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
Input #0, matroska,webm, from 'file:/video/Movies/movie.mkv':
  Metadata:
    encoder         : libmakemkv v1.10.0 (1.3.3/1.4.4) darwin(x86-release)
    creation_time   : 2016-07-27T11:53:20.000000Z
  Duration: 01:10:00.60, start: 0.000000, bitrate: 1896 kb/s
    Chapter #0:0: start 0.000000, end 300.300000

```
Finally trying the line directly within docker gives the following:

```
ffmpeg version 4.0.3 Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.3.0-27ubuntu1~18.04)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-vaapi --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gnutls --enable-gpl --enable-libass --enable-libfreetype --enable-libvidstab --enable-libmp3lame --enable-libopenjpeg --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx265 --enable-libxvid --enable-libx264 --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-postproc --enable-cuvid --enable-nvenc --enable-version3 --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib --extra-libs=-ldl --prefix=/opt/ffmpeg
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
Input #0, matroska,webm, from 'file:/video/Movies/movie.mkv':
  Metadata:
    encoder         : libmakemkv v1.10.0 (1.3.3/1.4.4) darwin(x86-release)
    creation_time   : 2016-07-27T11:53:20.000000Z
  Duration: 01:10:00.60, start: 0.000000, bitrate: 1896 kb/s
    Chapter #0:0: start 0.000000, end 300.300000
    Metadata:
      title           : Chapter 01
    Chapter #0:1: start 300.300000, end 600.099500
    Metadata:
      title           : Chapter 02
    Chapter #0:2: start 600.099500, end 900.399500
    Metadata:
      title           : Chapter 03
    Chapter #0:3: start 900.399500, end 1200.199000
    Metadata:
      title           : Chapter 04
    Chapter #0:4: start 1200.199000, end 1500.499000
    Metadata:
      title           : Chapter 05
    Chapter #0:5: start 1500.499000, end 1800.298500
    Metadata:
      title           : Chapter 06
    Chapter #0:6: start 1800.298500, end 2100.098000
    Metadata:
      title           : Chapter 07
    Chapter #0:7: start 2100.098000, end 2400.398000
    Metadata:
      title           : Chapter 08
    Chapter #0:8: start 2400.398000, end 2700.197500
    Metadata:
      title           : Chapter 09
    Chapter #0:9: start 2700.197500, end 3000.497500
    Metadata:
      title           : Chapter 10
    Chapter #0:10: start 3000.497500, end 3300.297000
    Metadata:
      title           : Chapter 11
    Chapter #0:11: start 3300.297000, end 3600.096500
    Metadata:
      title           : Chapter 12
    Chapter #0:12: start 3600.096500, end 3900.396500
    Metadata:
      title           : Chapter 13
    Chapter #0:13: start 3900.396500, end 4200.596400
    Metadata:
      title           : Chapter 14
    Stream #0:0(eng): Video: mpeg2video (Main), yuv420p(tv, progressive), 720x480 [SAR 8:9 DAR 4:3], 29.97 fps, 29.97 tbr, 1k tbn, 59.94 tbc
    Metadata:
      BPS-eng         : 1698436
      DURATION-eng    : 01:10:00.596400000
      NUMBER_OF_FRAMES-eng: 125892
      NUMBER_OF_BYTES-eng: 891805810
      SOURCE_ID-eng   : 0100E0
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.10.0 darwin(x86-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2016-07-27 11:53:20
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
    Stream #0:1(eng): Audio: ac3, 48000 Hz, stereo, fltp, 192 kb/s (default)
    Metadata:
      title           : Stereo
      BPS-eng         : 192000
      DURATION-eng    : 01:10:00.608000000
      NUMBER_OF_FRAMES-eng: 131269
      NUMBER_OF_BYTES-eng: 100814592
      SOURCE_ID-eng   : 0180BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.10.0 darwin(x86-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2016-07-27 11:53:20
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
Stream mapping:
  Stream #0:0 -> #0:0 (mpeg2video (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (ac3 (native) -> aac (native))
Press [q] to stop, [?] for help
[h264_vaapi @ 0x55e46d79c080] Encoding entrypoint not found (7 / 6).
Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
[aac @ 0x55e46d79aa00] Qavg: 662.453
[aac @ 0x55e46d79aa00] 2 frames left in the queue on closing
Conversion failed!
```

**Comment by JustAMan** [comment]
@GoodOmens83,
Please check if `jellyfin` user within the container can access vaapi device file (i.e. it most likely has to be in a correct group, or a device has to be world-writable).

@An4cl3t0,
The original issue seems to be caused by outdated ffmpeg. Could you please update to Jellyfin 10.4.0 (and ffmpeg 4.2.1) and see if it fixes your issue?

**Comment by An4cl3t0** [comment]
@JustAMan, 
resolved in Jellyfin 10.4.0 as you said. Thanks.
